niceman/distributions/debian.py

```diff
@@ -354,11 +356,20 @@ def identify_distributions(self, files):
         yield dist, remaining_files
 
     def _get_packagefields_for_files(self, files):
-        # Define a function to process batches of files and return
-        # a dictionary that maps files to packages
-        def proc_dpkg_query(subfiles, file_to_package_dict):
-            out = self._run_dpkg_query(subfiles)
-
+        # Call dpkg query in batches
+        exec_gen = execute_command_batch(self._session,
+                                         ['dpkg-query', '-S'],
+                                         files,
+                                         CommandError)
+        # Parse and accumulate stat results in a dict
+        file_to_package_dict = {}
+        for (out, _, exc) in exec_gen:
+            if exc:
+                stderr = utils.to_unicode(exc.stderr, "utf-8")
+                if 'no path found matching pattern' in stderr:
+                    out = exc.stdout  # One file not found, so continue
+                else:
+                    raise exc  # some other fault -- handle it above
```

Comment:
Eh -- my bad I think -- we wouldn't be to reraise the exception this way -- it would loose original exception throwing location:
```shell
$> nl -b a /tmp/1.py
     1	
     2	def fun():
     3	  raise Exception("buga")
     4	
     5	try:
     6	   fun()
     7	except Exception as exc:
     8	   vexc = exc
     9	
    10	# now we are here "reraising" it and information will be local
    11	raise vexc

$> python /tmp/1.py
Traceback (most recent call last):
  File "/tmp/1.py", line 11, in <module>
    raise vexc
Exception: buga
```
correct way would  be for us  to return exception info so it could be "properly" reraised:
```shell
$> nl -b a /tmp/1proper.py 
     1	import sys        
     2	from six import reraise
     3	
     4	def fun():
     5	  raise Exception("buga")
     6	
     7	try:
     8	   fun()
     9	except Exception as exc:
    10	   exc_info = sys.exc_info()
    11	
    12	# now we are here "reraising" it and information will be local
    13	reraise(*exc_info)

$> python /tmp/1proper.py 
Traceback (most recent call last):
  File "/tmp/1proper.py", line 13, in <module>
    reraise(*exc_info)
  File "/tmp/1proper.py", line 8, in <module>
    fun()
  File "/tmp/1proper.py", line 5, in fun
    raise Exception("buga")
Exception: buga
```
The `[1]` of that tuple is the exception itself, so you could access its attributes (e.g., `.stderr`).
feel free to postpone this to a separate PR

Reply:
It appears that raising an exception properly with past history changed from Python 2 to 3. However, if I update command_batch to be more explicit in what exceptions we accept (specifically by passing in a filter-like function), then we can have the generator throw those we don't accept.  The generator will then be able to use the no argument raise keyword to pass the exception gracefully.

Let me give this a go and see what you think.

Reply:
I'm still not resolving all the issues, but I committed the fix to address the handling of exceptions.

Reply:
> It appears that raising an exception properly with past history changed from Python 2 to 3.

Doesn't `six.reraise` let use work around this?

Reply:
Nice.  I didn't see that.  My current approach avoids the issue (as long as ya'll are OK with it).